### PR TITLE
Add Granular Services Counts to Telemetry

### PR DESCRIPTION
**Pull request by AlexFenlon**
### Proposed changes

This change granulaises the counting of services so instead of a total number, this adds a count for each service 
 eg. ClusterIP, LoadBalancer, NodePort and ExternalName. 

I have added go tests for this functionality.

```shell
kubectl get services -A
NAMESPACE     NAME                                  TYPE           CLUSTER-IP       EXTERNAL-IP                                                     PORT(S)                      AGE
default       coffee-svc                            ClusterIP      10.99.38.53      <none>                                                          80/TCP                       91m
default       externalname-service                  ExternalName   <none>           secure-app-external-backend-svc.external-ns.svc.cluster.local   <none>                       45m
default       kubernetes                            ClusterIP      10.96.0.1        <none>                                                          443/TCP                      92m
default       my-release-nginx-ingress-controller   LoadBalancer   10.111.0.251     <pending>                                                       80:30572/TCP,443:31414/TCP   2m24s
default       tea-svc                               ClusterIP      10.103.202.113   <none>                                                          80/TCP                       91m
external-ns   secure-app-external-backend-svc       ClusterIP      10.111.84.145    <none>                                                          8443/TCP                     45m
kube-system   kube-dns                              ClusterIP      10.96.0.10       <none>                                                          53/UDP,53/TCP,9153/TCP       92m
```

Here is a snippet of the log 
```shell
Secrets:1 ClusterIPServices:5 NodePortServices:0 LoadBalancerServices:1 ExternalNameServices:1
```

Resolves: #5452

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by jjngx — APPROVED** [review]
🚀

File: docs/content/overview/product-telemetry.md [hunk 1]
```diff
@@ -34,7 +34,9 @@ These are the data points collected and reported by NGINX Ingress Controller:
 - **TransportServers** The number of TransportServer resources managed by NGINX Ingress Controller.
 - **Replicas** Number of Deployment replicas, or Daemonset instances.
 - **Secrets** Number of Secret resources managed by NGINX Ingress Controller.
-- **Services** Number of Services referenced by VirtualServers, VirtualServerRoutes, TransportServers and Ingresses.
+- **ClusterIPServices** Number of ClusterIP Services managed by NGINX Ingress Controller.
+- **NodePortServices** Number of NodePort Services managed by NGINX Ingress Controller.
+- **ExternalNameServices** Number of ExternalName Services managed by NGINX Ingress Controller.
```
Comment (vepatel) [inline docs/content/overview/product-telemetry.md:39]: @AlexFenlon we should add service type [LoadBalancer](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer) as well

Comment (AlexFenlon) [reply docs/content/overview/product-telemetry.md:39]: @j1m-ryan actually asked me to remove this. I had LoadBalancer here initially.

Comment (vepatel) [reply docs/content/overview/product-telemetry.md:39]: is that because its cloud provider specific not core to k8s?

Comment (j1m-ryan) [reply docs/content/overview/product-telemetry.md:39]: i misunderstood the pr, you can add it back

File: internal/configs/configurator.go [hunk 2]
```diff
@@ -1604,106 +1604,6 @@ func (cnf *Configurator) getMinionIngressAnnotations(annotationSet map[string]bo
 	return annotationSet
 }
 
-// GetServiceCount returns the total number of unique services referenced by Ingresses, VS's, VSR's, and TS's
-func (cnf *Configurator) GetServiceCount() int {
```
[the remark this answers is not in the document]
Comment (AlexFenlon) [reply internal/configs/configurator.go:1608]: The functionality for thesd were no longer needed, these functions were just used to get the number of services eg. namespace/tea-svc and namespace/coffee-svc and just gets the total len of these. So my implementation is as the title suggests, more granular and specifies the number of service types.  `GetServiceCount()` just gets the total number of services.

**Review by shaun-nx — APPROVED** [review]

**Pull request merged**